src/test/java/org/highj/util/Gen.java

```diff
@@ -0,0 +1,29 @@
+package org.highj.util;
+
+import org.highj.data.List;
+import org.highj.data.Stream;
+
+import java.util.Random;
+import java.util.function.Function;
+
+public interface Gen<A> {
+
+    Random rnd = new Random();
```

Comment:
I think a mutable ```Gen``` is actually OK if used for local mutation. ```assertThat``` is mutable too, we are going to have to do some sort of mutation at some level.